### PR TITLE
changes made in Login and user repo

### PR DESCRIPTION
**Pull request by nishanth1697**
changes made in Login and user repo

### CODE REVIEW

File: Maxis.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26228.4
+# Visual Studio 14
+VisualStudioVersion = 14.0.25420.1
```
Comment (kamalakar1981) [inline Maxis.sln]: revert this file.

File: Maxis/Controllers/LoginController.cs [hunk 2]
```diff
@@ -12,78 +10,51 @@ namespace Maxis.Controllers
...
-                        Username = userName
-                    });
-                return Json(roles, JsonRequestBehavior.AllowGet);
-            }
-            catch (Exception ex)
-            {
-                return Json(ex.Message.ToString(), JsonRequestBehavior.AllowGet);
-            }
-           
+            ValidateUser(loginModel);
```
Comment (kamalakar1981) [inline Maxis/Controllers/LoginController.cs]: what does this method return, how are you validating the user can login or not.

File: Maxis/Infrastructure/Repositories/UserRepository.cs [hunk 3]
```diff
@@ -1,168 +1,184 @@
...
+                            Title = ep.Title,
+                            Status = ep.Status,
+                            Roles = e.RoleName
+                        }).Single();
+            return user;
+        }
+
+        public static string Encrypt(string password)
+        {
+            password = password ?? "";
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: why do you need this line?

File: Maxis/Infrastructure/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -1,168 +1,184 @@
...
+            var encoding = new System.Text.ASCIIEncoding();
+            var res = "";
+            var keyByte = encoding.GetBytes(password);
+            var messageBytes = encoding.GetBytes(ConfigurationManager.AppSettings["Hashkey"]);
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: salt key should not be same for all user, new key should be generated for each user.

File: Maxis/Infrastructure/Repositories/UserRepository.cs [hunk 5]
```diff
@@ -1,168 +1,184 @@
...
+                            Roles = e.RoleName
+                        }).Single();
+            return user;
+        }
+
+        public static string Encrypt(string password)
+        {
+            password = password ?? "";
+            var encoding = new System.Text.ASCIIEncoding();
+            var res = "";
```
Comment (kamalakar1981) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: use meaningful name for variable

File: Maxis/Interfaces/ICustomPrincipal.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Maxis.Interfaces
+{
+    interface ICustomPrincipal : IPrincipal
+    {
+        int UserId { get; set; }
```
Comment (kamalakar1981) [inline Maxis/Interfaces/ICustomPrincipal.cs]: why do you need this interface ?

File: Maxis/Interfaces/CustomPrincipal.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        public string Department { get; set; }
+        public string Title { get; set; }
+        public string Status { get; set; }
+        public int RoleId { get; set; }
+    }
+
+    public class CustomPrincipalSerializeModel
+    {
+        public int UserId { get; set; }
+        public string Username { get; set; }
```
Comment (kamalakar1981) [inline Maxis/Interfaces/CustomPrincipal.cs]: remove unused properties & give meaningful name to the class

File: Maxis/ViewModels/CustomPrincipal.cs [hunk 8]
```diff
@@ -1,7 +1,4 @@
-﻿
-using System.Security.Principal;
-
-
+﻿using System.Security.Principal;
 namespace Maxis.ViewModels
 {
     public class CustomPrincipal : IPrincipal
```
Comment (kamalakar1981) [inline Maxis/ViewModels/CustomPrincipal.cs]: why there are duplicate model classes please clean up

File: Maxis/Infrastructure/Repositories/UserRepository.cs (hunk 3 above)
Comment (nishanth1697) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: Removed the line.

File: Maxis/Infrastructure/Repositories/UserRepository.cs (hunk 4 above)
Comment (nishanth1697) [inline Maxis/Infrastructure/Repositories/UserRepository.cs]: This i would be implementing after the check in as i need to do a bit of R&D.

File: Maxis/Controllers/LoginController.cs (hunk 2 above)
Comment (nishanth1697) [inline Maxis/Controllers/LoginController.cs]: This check is done by LDAP server code.This is not implemented in the code as we dont have acces to that server from our domain.

File: Maxis/Interfaces/CustomPrincipal.cs (hunk 7 above)
Comment (nishanth1697) [inline Maxis/Interfaces/CustomPrincipal.cs]: Done the same.

File: Maxis/ViewModels/CustomPrincipal.cs (hunk 8 above)
Comment (nishanth1697) [inline Maxis/ViewModels/CustomPrincipal.cs]: Cleaned up the duplicated model.

**Pull request merged**